### DISCUSSION TITLE
how to import ethereum wallet by just using private key in android?

### DISCUSSION THREAD

**Discussion by yadavbipin** [web3j/web3j#1390] [General]
In web3j methods are been provided to import a wallet using Keystore JSON file or mnemonic but when I try to import my wallet using mnemonic seed different address is generated  and the methods provided are
The WalletUtils class can be used to load your credentials from various formats:

WalletUtils.loadCredentials(String password,String source)
WalletUtils.loadBip39Credentials(String password,String mnemonic)
WalletUtils.loadBip39Credentials(String password,String mnemonic)
WalletUtils.loadJsonCredentials(String password,String content)

which are for importing from Keystore JSON file or mnemonic what if I want to import my wallet just by using private key?

**Comment by mohamedelshami** [comment]
Credentials instance can be created using factory method:

`Credentials.create("<PRIVATE_KEY>")`
